src/iceberg/snapshot.h

```diff
@@ -0,0 +1,217 @@
+/// \brief Total number of live positional/equality delete files and deletion vectors in
+/// the snapshot
+constexpr std::string_view kTotalDeleteFilesKey = "total-delete-files";
+/// \brief Number of records added in the snapshot
+constexpr std::string_view kAddedRecordsKey = "added-records";
+/// \brief Number of records deleted in the snapshot
+constexpr std::string_view kDeletedRecordsKey = "deleted-records";
+/// \brief Total number of records in the snapshot
+constexpr std::string_view kTotalRecordsKey = "total-records";
+/// \brief The size of files added in the snapshot
+constexpr std::string_view kAddedFileSizeKey = "added-files-size";
+/// \brief The size of files removed in the snapshot
+constexpr std::string_view kRemovedFileSizeKey = "removed-files-size";
+/// \brief Total size of live files in the snapshot
+constexpr std::string_view kTotalFileSizeKey = "total-files-size";
+/// \brief Number of position delete records added in the snapshot
+constexpr std::string_view kAddedPosDeletesKey = "added-position-deletes";
+/// \brief Number of position delete records removed in the snapshot
+constexpr std::string_view kRemovedPosDeletesKey = "removed-position-deletes";
+/// \brief Total number of position delete records in the snapshot
+constexpr std::string_view kTotalPosDeletesKey = "total-position-deletes";
+/// \brief Number of equality delete records added in the snapshot
+constexpr std::string_view kAddedEqDeletesKey = "added-equality-deletes";
+/// \brief Number of equality delete records removed in the snapshot
+constexpr std::string_view kRemovedEqDeletesKey = "removed-equality-deletes";
+/// \brief Total number of equality delete records in the snapshot
+constexpr std::string_view kTotalEqDeletesKey = "total-equality-deletes";
+/// \brief Number of duplicate files deleted (duplicates are files recorded more than once
+/// in the manifest)
+constexpr std::string_view kDeletedDuplicatedFilesKey = "deleted-duplicate-files";
+/// \brief Number of partitions with files added or removed in the snapshot
+constexpr std::string_view kChangedPartitionCountProp = "changed-partition-count";
+
+/// Other Fields
+/// See https://iceberg.apache.org/spec/#other-fields
+
+/// \brief The Write-Audit-Publish id of a staged snapshot
+constexpr std::string_view kWAPIDKey = "wap.id";
+/// \brief The Write-Audit-Publish id of a snapshot already been published
+constexpr std::string_view kPublishedWAPIDKey = "published-wap-id";
+/// \brief The original id of a cherry-picked snapshot
+constexpr std::string_view kSourceSnapshotIDKey = "source-snapshot-id";
+/// \brief Name of the engine that created the snapshot
+constexpr std::string_view kEngineNameKey = "engine-name";
+/// \brief Version of the engine that created the snapshot
+constexpr std::string_view kEngineVersionKey = "engine-version";
+
+/// \brief The operation field is used by some operations, like snapshot expiration, to
+/// skip processing certain snapshots.
+enum class Operation {
```

Comment:
Same here - is there ever a chance we will have another "Operation"-like enum?

Reply:
In that case we need a separate namespace

Reply:
Is it acceptable by embedding the Operation in Summary class?